pilot/pkg/xds/debug.go

```
@@ -529,15 +530,25 @@ func (s *DiscoveryServer) adsz(w http.ResponseWriter, req *http.Request) {
 	if s.handlePushRequest(w, req) {
 		return
 	}
+	connections := s.Clients()
+	if proxyID := req.URL.Query().Get("proxyID"); proxyID != "" {
```

use `s.getDebugConnection` maybe to simplify?

---

We can modify it to make it not fail on missing proxyID query param if needed?